mito-ds/monorepo
# Graph Filtering 2.0

# Description

Update the filtering heuristic used by graphing in Mito and give users more control over when it is applied. The goal of this PR is to: Stop users from unknowingly graphing large datasets that will crash their browser tab. Note that the the filtering heuristic might seem overly aggressive, but the graphs become more intensive when we introduce the color property. Nonetheless, we will tweak the heuristic cutoff as we introduce the rest of the graphing parameters.  

# Testing

1. Test that if you have a script with the `filter_df_to_safe_size` function in it, that the code still functions. 
2. Make sure that when the filtering toggle is on, you can't crash your browser tab. 
3. Make sure that the toggle stays correctly configured when you open/close the graph, and that it resets when you switch the sheet you are graphing.

- [x] I have tested this on real data that is reasonable and large
- [x] If I changed the interaction with JupyterLab, I tested that it does not break other programs (like VS Code), and tested that it works "multiple times" in the same notebook.

# Documentation

Let's create a new video when the entire graphing redux is done. Not now
